provisioner/ansible/provisioner.go

```diff
@@ -855,15 +855,24 @@ func (p *Provisioner) executeAnsible(ui packersdk.Ui, comm packersdk.Communicato
 	// remove winrm password from command, if it's been added
 	flattenedCmd := strings.Join(cmd.Args, " ")
 	sanitized := flattenedCmd
-	winRMPass, ok := p.generatedData["WinRMPassword"]
-	if ok && winRMPass != "" {
-		sanitized = strings.Replace(sanitized,
-			winRMPass.(string), "*****", -1)
-	}
-	if checkArg("ansible_password", args) {
-		usePass, ok := p.generatedData["Password"]
-		if ok && usePass != "" {
-			sanitized = strings.Replace(sanitized, usePass.(string), "*****", -1)
+
+	for _, arg := range p.config.ExtraArguments {
+		args := strings.SplitN(arg, "=", 2)
+		if len(args) != 2 {
+			continue
+		}
+		if strings.Contains(strings.ToLower(args[0]), "password") ||
```

Comment:
If in the future if we find folks wanting to redact other secrets `API_TOKEN=xxxx` would it make sense for us to think about redacting all environment variables by default and maybe only printing the names?

Reply:
Or maybe providing an additional configuration option for setting secret variables which are part of the extra arguments that will be redacted by default.

Reply:
> would it make sense for us to think about redacting all environment variables by default and maybe only printing the names?

This is hard here because we can't know them, or this would require a big architectural change.

> Or maybe providing an additional configuration option for setting secret arguments which get added as extra arguments that will be redacted by default.

So normally the 'sensitive-variables' option allows to fix that; but because the ansible plugin does some funky stuff with our output system, sometimes Packer's ui can't redact these out. So I think one better way to do this would be to actually go and fix how the ui works and make something more consistent with other plugin that Packer core could obfuscate.

Reply:
>. So I think one better way to do this would be to actually go and fix how the ui works and make something more consistent with other plugin that Packer core could obfuscate.

Agreed